crates/oxc_minifier/src/peephole/fold_constants.rs

```diff
@@ -255,7 +254,46 @@ impl<'a> PeepholeOptimizations {
                 // non-nullish condition => this expression evaluates to the left side.
                 Some(logical_expr.left.take_in(ctx.ast.allocator))
             }
-            ValueType::Undetermined => None,
+            ValueType::Undetermined => match Self::is_nullish(left) {
+                Some(true) => Some(logical_expr.right.take_in(ctx.ast.allocator)),
+                Some(false) => Some(logical_expr.left.take_in(ctx.ast.allocator)),
+                None => None,
+            },
+        }
+    }
+
+    fn is_nullish(expr: &Expression<'a>) -> Option<bool> {
+        match expr.get_inner_expression() {
+            Expression::NullLiteral(_) => Some(true),
+            Expression::ImportExpression(_)
+            | Expression::BinaryExpression(_)
+            | Expression::UpdateExpression(_) => Some(false),
+            Expression::Identifier(id) => match id.name.as_str() {
+                "undefined" => Some(true),
```

Comment:
There are special methods for checking these, they can be global.

I can't recall what they are, try and find them yourself ;-)